File: 30 Days of Code/Day 11 2D Arrays/Solution.cpp
// https://www.hackerrank.com/challenges/30-2d-arrays/problem

#include <bits/stdc++.h>

using namespace std;



int main()
{
    vector<vector<int>> arr(6);
    for (int i = 0; i < 6; i++) {
        arr[i].resize(6);

        for (int j = 0; j < 6; j++) {
            cin >> arr[i][j];
        }

        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }

    vector<int>sum;
    for(int i = 0; i < 4; ++i){
        for(int j = 0; j < 4; ++j){
            int s = (arr[i][j] + arr[i][j+1] + arr[i][j+2]) + (arr[i+1][j+1]) + (arr[i+2][j] + arr[i+2][j+1]+arr[i+2][j+2]);
            sum.push_back(s);
        }
    }
    cout<<*max_element(sum.begin(), sum.end());
    return 0;
}

File: 30 Days of Code/Day 11 2D Arrays/Solution.py
# https://www.hackerrank.com/challenges/30-2d-arrays/problem

#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    arr = []
    total = []
    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))
    for i in range(4):
        for j in range(4):
            s = (arr[i][j]+arr[i][j+1]+arr[i][j+2]) + (arr[i+1][j+1]) + (arr[i+2][j]+arr[i+2][j+1]+arr[i+2][j+2])
            total.append(s)
    print(max(total))